http-common/src/connector.rs

```diff
@@ -173,7 +81,18 @@ impl Connector {
 	#[cfg(feature = "tokio02")]
 	pub async fn incoming(self) -> std::io::Result<Incoming> {
 		match self {
-			Connector::Fd { fd, .. } => {
+			Connector::Fd { fallback } => {
+				let fd = match get_systemd_socket() {
+					Ok(Some(fd)) => fd,
+
+					Ok(None) => {
+						let fallback_incoming: std::pin::Pin<Box<dyn std::future::Future<Output = _>>> = Box::pin(fallback.incoming());
+						return fallback_incoming.await;
```

Comment:
Why not just `return fallback.incoming().await;`?

Reply:
Recursive futures would be infinitely-sized.

Reply:
ahhhh, didn't see it was recursive.

I take it that's why you've got an explicit type annotation (to coerce it to a `dyn` Future)?

Reply:
Yeah.

`cert/aziot-certd/src/lib.rs` has another example of doing a recursive async fn in its `fn create_cert` with a different approach, but I think this approach is better. I might switch that one to this approach too if I ever edit it for another reason.